Userland/Libraries/LibIPC/Decoder.cpp

```diff
@@ -131,9 +131,14 @@ ErrorOr<Core::AnonymousBuffer> decode(Decoder& decoder)
         return Core::AnonymousBuffer {};
 
     auto size = TRY(decoder.decode_size());
-    auto anon_file = TRY(decoder.decode<IPC::File>());
 
-    return Core::AnonymousBuffer::create_from_anon_fd(anon_file.take_fd(), size);
+#ifndef AK_OS_WINDOWS
+    auto anon_file = TRY(decoder.decode<IPC::File>());
+    auto id = anon_file.take_fd();
+#else
+    auto id = TRY(decoder.decode<ByteString>());
+#endif
```

Comment:
here, I would rather update IPC::File to know whether it's holding a C/Unix `file descriptor` or a HANDLE and manage that internally, rather than having to add ifdefs for windows here in the decoder for anonymous buffer specifically.

Reply:
AnonymousBuffer on Windows doesn't use IPC::File, it just passes its name.